plugins/modules/k8s_taint.py

```diff
@@ -0,0 +1,279 @@
+
+
+def _get_difference(a, b):
+    return [
+        a_item for a_item in a if not any(_equal_dicts(a_item, b_item) for b_item in b)
+    ]
+
+
+def argspec():
+    argument_spec = copy.deepcopy(AUTH_ARG_SPEC)
+    argument_spec.update(
+        dict(
+            state=dict(type="str", choices=["present", "absent"], default="present"),
+            name=dict(type="str", required=True),
+            taints=dict(type="list", required=True, elements="dict"),
+            overwrite=dict(type="bool", default=False),
+        )
+    )
+
+    return argument_spec
+
+
+class K8sTaintAnsible:
+    def __init__(self, module):
+        self.module = module
+        self.k8s_ansible_mixin = K8sAnsibleMixin(module=self.module)
+        self.k8s_ansible_mixin.client = get_api_client(module=self.module)
+        self.k8s_ansible_mixin.module = self.module
+        self.k8s_ansible_mixin.argspec = self.module.argument_spec
+        self.k8s_ansible_mixin.check_mode = self.module.check_mode
+        self.k8s_ansible_mixin.params = self.module.params
+        self.k8s_ansible_mixin.fail_json = self.module.fail_json
+        self.k8s_ansible_mixin.fail = self.module.fail_json
+        self.k8s_ansible_mixin.exit_json = self.module.exit_json
+        self.k8s_ansible_mixin.warn = self.module.warn
+        self.k8s_ansible_mixin.warnings = []
+        self.api_instance = core_v1_api.CoreV1Api(self.k8s_ansible_mixin.client.client)
+        self.k8s_ansible_mixin.check_library_version()
+        self.changed = False
+
+    def get_node(self, name):
+        try:
+            node = self.api_instance.read_node(name=name)
+        except ApiException as exc:
+            if exc.reason == "Not Found":
+                self.module.fail_json(msg="Node '{0}' has not been found.".format(name))
+            self.module.fail_json(
+                msg="Failed to retrieve node '{0}' due to: {1}".format(
+                    name, exc.reason
+                ),
+                status=exc.status,
+            )
+        except Exception as exc:
+            self.module.fail_json(
+                msg="Failed to retrieve node '{0}' due to: {1}".format(
+                    name, to_native(exc)
+                )
+            )
+
+        return node
+
+    def patch_node(self, taints):
+        body = {"spec": {"taints": taints}}
+
+        try:
+            result = self.api_instance.patch_node(
+                name=self.module.params.get("name"), body=body
+            )
+        except Exception as exc:
+            self.module.fail_json(
+                msg="Failed to patch node due to: {0}".format(to_native(exc))
+            )
+
+        return result.to_dict()
+
+    def execute_module(self):
+        result = {"result": {}}
+        state = self.module.params.get("state")
+        taints = self.module.params.get("taints")
+        name = self.module.params.get("name")
+
+        node = self.get_node(name)
+        existing_taints = node.spec.to_dict()["taints"] or []
```

Comment:
```suggestion
        existing_taints = node.spec.to_dict().get("taints", [])
```
This is raising exception when the key does not exist

Reply:
Since the key always exists for the moment and if in a near future it won't be returned anymore, I did this for the moment.
`existing_taints = node.spec.to_dict().get("taints") or []`